soc/arm/st_stm32/stm32h7/Kconfig.series

```diff
@@ -12,7 +12,6 @@ config SOC_SERIES_STM32H7X
 	select HAS_STM32CUBE
 	select CPU_HAS_ARM_MPU
 	select HAS_SWO
-	select REQUIRES_FULL_LIBC
```

Comment:
Why is this required?

Reply:
1. otherwise `minlibc` can not be used and compiling of `samples/net/sockets/civetweb` produces following warning:
```
warning: MINIMAL_LIBC_MALLOC_ARENA_SIZE (defined at lib/libc/Kconfig:92) was assigned the value
'131072' but got the value ''
```
2. `soc/arm/st_stm32f7/stm32f7/Kconfig.series` - does not contain `REQUIRES_FULL_LIBC` also

Reply:
I 'm concerned about the impact of not selecting REQUIRES_FULL_LIBC?
Could it cause regression other tests or samples ?
I would prefer to wait for @erwango feedback

Reply:
> I 'm concerned about the impact of not selecting REQUIRES_FULL_LIBC?
> Could it cause regression other tests or samples ?
> I would prefer to wait for @erwango feedback

I would say - see 2. in my reply above. And also buildkite runs properly. But as you wish - we can wait for @erwango

Reply:
@ABOSTM @gmarull Plese see description of #27373 to compile `samples/net/sockets/civetweb` for one of stm32h7 boards. And you will see which impact explicit select of `REQUIRES_FULL_LIBC` has on this sample. I just followed same approach from stm32f7 series, where `REQUIRES_FULL_LIBC` does not selected explicitly (see description of #27373 for it also).

Reply:
> soc/arm/st_stm32f7/stm32f7/Kconfig.series - does not contain REQUIRES_FULL_LIBC also

That doesn't mean H7 don't require it. There may be other solutions that keep REQUIRES_FULL_LIBC and remove your errors ...
My understanding is that there are 2 issues: 
1) The warning which comes from the fact that the test defines CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE (prj.conf)  even when not needed (when REQUIRES_FULL_LIBC is defined). This could for example be solved by changing the sample.
2) The compilation error. This look to me like the test is not designed to work with the full LibC.
So now, 
- should we adapt the sample to work with both non-full and full libC? 
- exclude the board for this test (as it is not compatible with full libC)? 
- or change the driver to not use the Full LibC (like you proposed) ?
Honestly I don't know. And prefer to open the discussion.

> And you will see which impact explicit select of `REQUIRES_FULL_LIBC` has on this sample

My concern is not about this sample, but about the impact on all others samples and tests.
I want to avoid regressions.

Reply:
I think that the intent of this PR is to add RNG support, so this parameter should not be changed here. If this change is required for something else I think a new PR should be created with some context to better understand the implications.

Reply:
@gmarull  @ABOSTM @erwango  The commit which is the point of this discussion were removed from this PR. Another PR #27389 for fulllibc explicit select remove were opened.